docs/source/using/troubleshooting.adoc

```diff
@@ -113,6 +113,15 @@ cause passwords to fail.
 Workaround: When creating and entering passwords, wrap the string with
 single quotes in the following format: `<password>`
 
+NOTE: The characters `", $, ` and \` are still interpreted by the shell,
```

Comment:
@hferentschik Looks like I was not clear enough with the details. Let me try to explain with the logs trace and experiment which I did to find out why we need another level of escaping.

1. Start minishift with `'Password$$'` parameter.

```
$ ./minishift start --username username --password 'Password$$' --show-libmachine-logs --v=5
...
Using SSH private key: /Users/prkumar/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/prkumar/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo -E subscription-manager register --auto-attach --username username --password Password$$ 
SSH cmd err, output: exit status 70: Registering to: subscription.rhsm.redhat.com:443/subscription
Invalid username or password. To create a login, please visit https://www.redhat.com/wapps/ugc/register.html
```

Above you can see the way SSH command detect it and try to run it will see without single quoted password parameter which interpreted differently in bash, same as you also pointed out in the comment.

2. Since now we have the command and ssh key already configured to box so I just killed the minishift process and VM is still running to do some checks on ssh side.

On the host let's try to run same ssh command with different escaping methods.

```
$ /usr/bin/ssh -F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/prkumar/.minishift/machines/minishift/id_rsa -p 22 sudo -E subscription-manager register --auto-attach --username username --password 'Password$$'
subscription.rhsm.redhat.com:443/subscription
Invalid username or password. To create a login, please visit https://www.redhat.com/wapps/ugc/register.html

$ /usr/bin/ssh -F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/prkumar/.minishift/machines/minishift/id_rsa -p 22 sudo -E subscription-manager register --auto-attach --username username --password /''Password$$'/'
Registering to: subscription.rhsm.redhat.com:443/subscription
The system has been registered with ID: 981f2562-3a0f-44a7-ba40-a71ab3866b73

$ /usr/bin/ssh -F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/prkumar/.minishift/machines/minishift/id_rsa -p 22 sudo -E subscription-manager register --auto-attach --username username --password Password\$\$
Registering to: subscription.rhsm.redhat.com:443/subscription
The system has been registered with ID: 981f2562-3a0f-44a7-ba40-a71ab3866b73
```

Above you can see last 2 methods works for a special character, for 2nd method it will only work with `$` special character but 3rd method will work for each special shell character as long as we provide escape. 

> the statement "The characters ", $, and ` are still interpreted by the shell even when you wrap the string with single quotes" is, however, wrong.

I agree and I really want some good wording around it why we need it.